packages/test/test-service-load/src/runner.ts

```diff
@@ -152,23 +152,34 @@ function* factoryPermutations<T extends IDocumentServiceFactory>(create: () => T
 		// Certain behavior (like driver caches) are per factory instance, and by reusing it we hit those code paths
 		// At the same time we want to test newly created factory.
 		let documentServiceFactory: T = factoryReused;
-		let headers: IRequestHeader = {};
+		let headers: IRequestHeader = { deltaConnection: "none" };
 		switch (counter % 5) {
 			default:
 			case 0:
 				documentServiceFactory = create();
 				break;
 			case 1:
-				headers = { [LoaderHeader.loadMode]: { opsBeforeReturn: "cached" } };
+				headers = {
```

Comment:
rather than hard coding these, you could use the pairwise generator to generate the combinations.

Reply:
maybe something like
```
function* factoryPermutations<T extends IDocumentServiceFactory>(create: () => T) {
	let counter = 0;
	const factoryReused = create();
	const loadModes = generatePairwiseOptions<IContainerLoadMode & { reuseFactory: boolean }>({
		reuseFactory: [true, false],
		opsBeforeReturn: [undefined, "cached", "all"],
		deltaConnection: [undefined, "none", "delayed"],
		pauseAfterLoad: [undefined],
	});
	while (true) {
		counter++;
		const loadMode = loadModes[counter % loadModes.length];

		// Switch between creating new factory vs. reusing factory.
		// Certain behavior (like driver caches) are per factory instance, and by reusing it we hit those code paths
		// At the same time we want to test newly created factory.
		const documentServiceFactory: T = loadMode.reuseFactory ? factoryReused : create();
		const headers: IRequestHeader = { [LoaderHeader.loadMode]: loadMode };
		yield { documentServiceFactory, headers };
	}
}

```